src/databricks/labs/remorph/assessments/pipeline.py

```diff
@@ -56,31 +59,57 @@ def _execute_sql_step(self, step: Step):
         credential_config = str(cred_file("remorph"))
 
-        try:
-            result = subprocess.run(
-                ["python", step.extract_source, "--db-path", db_path, "--credential-config-path", credential_config],
-                check=True,
-                capture_output=True,
-                text=True,
-            )
-
+        # Create a temporary directory for the virtual environment
+        with tempfile.TemporaryDirectory() as temp_dir:
+            venv_dir = Path(temp_dir) / "venv"
+            venv.create(venv_dir, with_pip=True)
+            venv_python = venv_dir / "bin" / "python"
+            venv_pip = venv_dir / "bin" / "pip"
+
+            # Install dependencies in the virtual environment
+            if step.dependencies:
+                logging.info(f"Installing dependencies: {', '.join(step.dependencies)}")
+                try:
+                    run([str(venv_pip), "install"] + step.dependencies, check=True, capture_output=True, text=True)
+                except CalledProcessError as e:
+                    logging.error(f"Failed to install dependencies: {e.stderr}")
+                    raise RuntimeError(f"Failed to install dependencies: {e.stderr}") from e
+
+            # Execute the Python script using the virtual environment's Python interpreter
             try:
-                output = json.loads(result.stdout)
-                if output["status"] == "success":
-                    logging.info(f"Python script completed: {output['message']}")
-                else:
-                    raise RuntimeError(f"Script reported error: {output['message']}")
-            except json.JSONDecodeError:
-                logging.info(f"Python script output: {result.stdout}")
-
-        except subprocess.CalledProcessError as e:
-            error_msg = e.stderr
-            logging.error(f"Python script failed: {error_msg}")
-            raise RuntimeError(f"Script execution failed: {error_msg}") from e
+                result = run(
+                    [
+                        str(venv_python),
+                        str(step.extract_source),
+                        "--db-path",
+                        db_path,
+                        "--credential-config-path",
+                        credential_config,
+                    ],
+                    check=True,
+                    capture_output=True,
+                    text=True,
+                )
+
+                try:
+                    output = json.loads(result.stdout)
+                    logger.debug(output)
+                    if output["status"] == "success":
+                        logging.info(f"Python script completed: {output['message']}")
+                    else:
+                        raise RuntimeError(f"Script reported error: {output['message']}")
+                except json.JSONDecodeError:
+                    logging.info(f"Python script output: {result.stdout}")
+
+            except CalledProcessError as e:
```

Comment:
It looks like upon failure we drop anything that was written to stdout. Do you think it's useful to log that?

Reply:
the idea the scripts being executed will pass down any RuntimeErrors error back to the caller in structured format.